### PR TITLE
Change how test assemblies opt-in to LibraryImportGenerator usage

### LINKED ISSUE #92655: Interop source generators failing build in dotnet/runtime when latest SDK is installed
### Description

When building for older frameworks the latest interop source generator is used, which ends up forcing the latest shared library `Microsoft.Interop.SourceGeneration.dll` while still using older source generators.  This makes the older source generators break if there are breaking changes to the internal `Microsoft.Interop.SourceGeneration.dll`.

### Reproduction Steps

Build C:\src\dotnet\runtime\src\libraries\Common\tests\TestUtilities\TestUtilities.csproj

### Expected behavior

Builds successfully

### Actual behavior

Build fails with
> CSC : error CS8784: Generator 'VtableIndexStubGenerator' failed to initialize. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'TypeLoadException' with message 'Could not load type 'Microsoft.Interop.TargetFramework' from assembly 'Microsoft.Interop.SourceGeneration, Version=42.42.42.42, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.' 

### Regression?

Yes, 

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

I attempted to fix this in https://github.com/dotnet/runtime/pull/92648 before fully understanding the problem.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Change test and test support assemblies to be auto-opted in to referencing LibraryImportGenerator instead of manually enabling it.

By changing this to an auto-opt-in, our TFM-based filtering logic kicks in and ensures that each project references the version of LibraryImportGenerator that corresponds to its TFM (as intended).

Fixes https://github.com/dotnet/runtime/issues/92655

### CODE REVIEW

File: eng/generators.targets [hunk 1]
```diff
@@ -8,14 +8,18 @@
...
          - doesn't reference the live targeting pack (i.e. when inbox) and
            - references System.Private.CoreLib, or
            - references System.Runtime.InteropServices -->
     <EnabledGenerators Include="LibraryImportGenerator"
                        Condition="'$(EnableLibraryImportGenerator)' == '' and
-                                  '$(IsSourceProject)' == 'true' and
+                                  (
+                                    '$(IsSourceProject)' == 'true' or
+                                    '$(IsTestProject)' == 'true' or
+                                    '$(IsTestSupportProject)' == 'true'
```
Comment (ViktorHofer) [inline eng/generators.targets:21]: This is either true for projects that target a non .NETCoreApp TFM or ones that disable implicit framework references and then reference System.Runtime.InteropServices or CoreLib. So essentially this logic only applies to our .NET Framework TFMs in test projects.

Is that intention? Just want to better understand the change.

Comment (jkoritzinsky) [reply eng/generators.targets:21]: Yes, this should only be true for .NET Framework or Standard TFMs in tests.

**Review by ViktorHofer — APPROVED** [review]

**Review by ericstj — APPROVED** [review]
I pulled this down and confirm it fixes https://github.com/dotnet/runtime/issues/92655.  

> By changing this to an auto-opt-in, our TFM-based filtering logic kicks in and ensures that each project references the version of LibraryImportGenerator that corresponds to its TFM (as intended).

Confirmed that this is happening.  Indeed, this is much better behavior.  I'm glad we're not depending on using the newer source generator when targeting older frameworks which carry it.

Thank you @jkoritzinsky

**Review by jtschuster — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Wasm failures are specific to this PR. Need to figure out what's causing it.

**Comment by jkoritzinsky** [comment]
Okay for some reason I can't find any other CI run that has these particular jobs running. I'm going to file a known issue.

**Comment by radical** [comment]
> Wasm failures are specific to this PR. Need to figure out what's causing it.

This fixes the issue - [`c40b2dc` (#92741)](https://github.com/dotnet/runtime/pull/92741/commits/c40b2dc683faea95b846db588763aea025adade3) which got introduced here.

**Pull request merged**